crates/wdk-sys/src/types.rs

```diff
@@ -38,6 +38,13 @@
     ),
     allow(clippy::pub_underscore_fields)
```

Comment:
Could we remove the old blocks?

Reply:
I'm not sure what you're referring to when you say old blocks. The change here is that the nightly version of the rust compiler added some new lints that are triggering on generated code. This block allows an exception for the new lint only in the compiler versions that are newer than the nightly version that introduced the lint. This is required so that older compiler versions won't fail build because of an unknown lint (since older compilers didn't have the lint). Our CI runs builds for stable, beta, and nightly builds so that any possible issues in upcoming rust versions are caught and addressed early.

Since these are triggering in generated code, I could actually just add a blanket exception for all the generated code, but I've chosen to keep it one by one since sometimes the lints can be avoided by tweaking a configuration option in bindgen to improve the quality of the generated code. I've found these nightly breakages to be fairly low effort to address and stay on top of, so I would prefer to continue doing it this way for now